src/Authoring/WinRT.SourceGenerator/DiagnosticHelpers.cs

```diff
@@ -330,6 +353,111 @@ private string SimplifySyntaxTypeString(string syntaxType)
             }
         }
 
+        private SpecialType[] AllowedStructFieldTypes = new SpecialType[] 
+        {
+            SpecialType.System_Boolean, 
+            SpecialType.System_String,
+            SpecialType.System_Single,
+            SpecialType.System_Double,
+            SpecialType.System_UInt16,
+            SpecialType.System_UInt32,
+            SpecialType.System_UInt64,
+            SpecialType.System_Int16,
+            SpecialType.System_Int32,
+            SpecialType.System_Int64,
+            SpecialType.System_Enum,
+        };
+
+        private static readonly Dictionary<string, List<string>> MappedCSharpTypeMethods = new Dictionary<string, List<string>>()
+        {
+            { "System.DateTimeOffset", new List<string> { "Add", "AddDays", "AddHours", "AddMilliseconds", "AddMinutes", "AddMonths", "AddSeconds", "AddTicks", "AddYears", "Compare", "CompareTo",

```

Comment:
You only need the interfaces as those are the ones that can be implemented by component classes.  The custom mapped classes I believe are all sealed so they are not used for extending.